at_secondary/at_secondary_server/lib/src/verb/handler/scan_verb_handler.dart

```diff
@@ -60,12 +60,19 @@ class ScanVerbHandler extends AbstractVerbHandler {
       if (forAtSign != null &&
           atConnectionMetadata.isAuthenticated &&
           forAtSign != currentAtSign) {
-        response.data =
-            await _getExternalKeys(forAtSign, scanRegex, atConnection);
+        if (page == null) {
```

Comment:
Even when the keys are external, that secondary's can should know how to give back results for a page. We should not get all of the results and then return a sublist.